File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/application/ports/SolicitacaoRepository.java
package br.ufpr.sept.so2.modules.solicitacoes.application.ports;

import br.ufpr.sept.so2.modules.solicitacoes.domain.Solicitacao;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;
import java.util.UUID;

public interface SolicitacaoRepository {

    Solicitacao save(Solicitacao solicitacao);

    Optional<Solicitacao> findById(UUID id);

    Optional<Solicitacao> findByProtocolo(String protocolo);

    Page<Solicitacao> findMinhas(
            UUID solicitanteId,
            String estado,
            String tipoCodigo,
            Integer ano,
            Pageable pageable
    );
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/application/ports/SolicitacaoRepository.kt
package br.ufpr.sept.so2.modules.solicitacoes.application.ports

import br.ufpr.sept.so2.modules.solicitacoes.domain.Solicitacao
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.Optional
import java.util.UUID

interface SolicitacaoRepository {

    fun save(solicitacao: Solicitacao): Solicitacao

    fun findById(id: UUID): Optional<Solicitacao>

    fun findByProtocolo(protocolo: String): Optional<Solicitacao>

    fun findMinhas(
        solicitanteId: UUID,
        estado: String?,
        tipoCodigo: String?,
        ano: Int?,
        pageable: Pageable,
    ): Page<Solicitacao>
}
